xwiki-commons-core/xwiki-commons-component/xwiki-commons-component-default/src/main/java/org/xwiki/component/annotation/DefaultComponentDependencyFactory.java

```diff
@@ -39,30 +39,43 @@ public class DefaultComponentDependencyFactory extends AbstractComponentDependen
     @Override
     public ComponentDependency createComponentDependency(Field field)
     {
-        DefaultComponentDependency dependency;
+        DefaultComponentDependency dependency = null;
 
-        Inject inject = field.getAnnotation(Inject.class);
-        if (inject != null) {
-            dependency = new DefaultComponentDependency();
+        if (field.getAnnotation(Inject.class) != null || field.getAnnotation(javax.inject.Inject.class) != null) {
+            dependency = createDependency(field);
+        } else {
+            dependency = null;
```

Comment:
This else doesn't seem necessary as the variable is already initialized with `null`. Also, isn't this a behavior change?